### PR TITLE
anuejn/version

### PR DESCRIPTION
**Pull request by anuejn**
- ✨ display current transcribee version in frontend
- 🐞 fix useStateDelayed rapid toggling
- 🐞 fix tooltip bottom arrow
- 🐞 close tooltip with delay

### CODE REVIEW

File: frontend/src/common/version.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        {versionText} + {diffShort}
+      </>
+    );
+  }
+
+  return (
+    <>
+      <div className="mb-10">
+        <div className="absolute bottom-0 h-10 text-xs left-1/2 -translate-x-1/2 opacity-50 hover:opacity-100 duration-400 transition-all">
+          Frontend build on {formatDate(version.date)}. {versionText}
```
Comment (pajowu) [inline frontend/src/common/version.tsx:54]:
```suggestion
          Frontend built on {formatDate(version.date)}. {versionText}
```

File: frontend/src/common/version.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <>
+      <Tooltip
+        className="inline-block underline decoration-dashed"
+        tooltipText={`until ${lastCommitOnMain.hash.substring(0, 10)} from ${formatDate(
+          lastCommitOnMain.date,
+        )}`}
+      >
+        {lastCommitOnMain.countSinceStart} commits on main
+      </Tooltip>
+    </>
```
Comment (pajowu) [inline frontend/src/common/version.tsx:23]: This could be a component

File: frontend/src/common/version.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  );
+
+  if (branch !== 'main') {
+    versionText = (
+      <>
+        {versionText} +{' '}
+        <Tooltip
+          className="inline-block underline decoration-dashed"
+          tooltipText={`until ${lastCommit.hash.substring(0, 10)} from ${formatDate(
+            lastCommit.date,
```
Comment (pajowu) [inline frontend/src/common/version.tsx:33]: I feel like this is not proper english?

**Review by pajowu — CHANGES_REQUESTED** [review]
The version is currently not shown on the document editor page. Is this on purpose? I'd prefer to have it there as well

File: frontend/src/common/version.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export function Version() {
+  const { lastCommitOnMain, branch, lastCommit, diffShort } = version;
+
+  let versionText = (
+    <>
+      <Tooltip
+        className="inline-block underline decoration-dashed"
+        tooltipText={`until ${lastCommitOnMain.hash.substring(0, 10)} from ${formatDate(
```
Comment (rroohhh) [inline frontend/src/common/version.tsx:17]: Maybe a weird idea, but making this a link to github with the correct revision would be fun.
We could even implement a way to download the source that way.

File: frontend/src/common/version.tsx (hunk 3 above)
Comment (anuejn) [reply frontend/src/common/version.tsx:33]: any suggestions?

**Review by pajowu — APPROVED** [review]

**Pull request merged**